operations/app/src/environments/tf

```diff
@@ -158,6 +167,69 @@ else
 fi
 ACTUAL_RESULTS="${ACTUAL_RESULTS?}${?}"
 
+RESOURCE_GROUP_NAME="$(echo "var.resource_group" | terraform console --var-file "${TFVARS?}" -var="terraform_caller_ip_address=${PUBLIC_IP_CIDR?}" | xargs)"
+RESOURCE_PREFIX="$(echo "var.resource_prefix" | terraform console --var-file "${TFVARS?}" -var="terraform_caller_ip_address=${PUBLIC_IP_CIDR?}" | xargs)"
+
+# Whitelist our IP to access firewalled resources
+function add_ip_whitelist() {
+    echo -e "${GRAY?}"
+    echo -e "Whitelisting IP for Azure resources:"
+
+    for resource in "${KeyVaults[@]}"; do
+        RESOURCE_NAME="${RESOURCE_PREFIX?}${resource?}"
+        echo -e "\t${RESOURCE_NAME?}..."
+        az keyvault network-rule add -g "${RESOURCE_GROUP_NAME?}" -n "${RESOURCE_NAME?}" --ip-address "${PUBLIC_IP_CIDR?}" > /dev/null 2>&1
+    done
+
+    for resource in "${StorageAccounts[@]}"; do
+        RESOURCE_NAME="${RESOURCE_PREFIX?}${resource?}"
+        echo -e "\t${RESOURCE_NAME?}..."
+        az storage account network-rule add -g "${RESOURCE_GROUP_NAME?}" -n "${RESOURCE_NAME?}" --ip-address "${PUBLIC_IP?}" > /dev/null 2>&1
+    done
+
+    for resource in "${!FunctionApps[@]}"; do
+        RESOURCE_NAME="${RESOURCE_PREFIX?}${resource?}"
+        IFS=','; read -ra FUNCTION_SLOTS <<< "${FunctionApps[${resource?}]}"
+        echo -e "\t${RESOURCE_NAME?}..."
+        for slot in "${FUNCTION_SLOTS[@]}"; do
+            az functionapp config access-restriction add -g "${RESOURCE_GROUP_NAME?}" -n "${RESOURCE_NAME?}" --slot "${slot?}" --rule-name "tf-${PUBLIC_IP_HASH}" --action Allow --ip-address "${PUBLIC_IP_CIDR?}" --priority 750 > /dev/null 2>&1
```

Comment:
Function Apps require a rule name. A sha1 hash of the IP address is used to avoid collisions.